### ISSUE TITLE
Received unhandled ObjectiveC exception

### ISSUE THREAD

**Issue by MartinDamov** [alexrainman/CarouselView#260]
Xamarin.Forms version - 2.3.4.247
CarouselView version - 4.4.3

I wanted to create an "infinite" carousel view. First I tried by having 3 views and when user swipes, I get the position, disable the animation, move back to the middle page (by setting the Position propertyo to 1) and then enable again the animation. This however didn't always update the UI on iOS. 

The next strategy I tried is by change the ObservableCollection as in the code below:

`       private ObservableCollection<Label> _views;
        private CarouselViewControl _carouselView;
        private int _currentPage = 0;

        public MainPage()
        {
            InitializeComponent();
            AddTestData();
            ParentLayout.Children.Add(_carouselView);
        }

        private void AddTestData()
        {
            _views = new ObservableCollection<Label>
            {
                _left,
                _middle,
                _right
            };

            _carouselView = new CarouselViewControl
            {
                BackgroundColor = Color.BurlyWood,
                VerticalOptions = LayoutOptions.FillAndExpand,
                HorizontalOptions = LayoutOptions.FillAndExpand,
                Position = 1,
                ItemsSource = _views,
                InterPageSpacing = 10
            };
            _carouselView.PositionSelected += OnPositionSelected;
            _carouselView.ItemTemplate = new DataTemplate(() => _views[_currentPage]);
        }

        private void OnPositionSelected(object sender, int position)
        {
            if (position.Equals(0))
            {
                Debug.WriteLine("CarouselView Position = 0");
                _currentPage = 0;
                var tmp = _views[2];
                _views.RemoveAt(2);
                _views.Insert(0, tmp);
            }
            else if (position.Equals(2))
            {
                Debug.WriteLine("CarouselView Position = 2");
                _currentPage = 2;
                var tmp = _views[0];
                _views.RemoveAt(0);
                _views.Insert(2, tmp);
            }
        }`

This works perfect on Android, but on iOS I get the following error when swipping: 

CarouselViewTest.iOS[251:86484] Xamarin.iOS: Received unhandled ObjectiveC exception: UIViewControllerHierarchyInconsistency A view can only be associated with at most one view controller at a time! View <Xamarin_Forms_Platform_iOS_Platform_DefaultRenderer: 0x1460f820; frame = (0 26.5; 320 541.5); autoresize = LM+W+RM+TM+H+BM; layer = <CALayer: 0x14610470>> is associated with <CarouselView_FormsPlugin_iOS_ViewContainer: 0x1460f900>. Clear this association before associating this view with <CarouselView_FormsPlugin_iOS_ViewContainer: 0x146056d0>.

Another issue I noticied is that on iOS the PositionChanged event is trigged 3 times (see console output below) while on Android it gets called only once (as supposed). 

Console output from iOS:

2017-11-16 09:33:33.604 CarouselViewTest.iOS[5130:1234592] ChildViewControllers count = 1
2017-11-16 09:33:41.089 CarouselViewTest.iOS[5130:1234592] ChildViewControllers count = 2
2017-11-16 09:33:41.091 CarouselViewTest.iOS[5130:1234592] ChildViewControllers count = 3
Carousel View Control pos = 2
Carousel View Control pos = 2
Carousel View Control pos = 2

I am using this plugin in other places and works perfectly both Android and iOS (I am not targeting UWP).

**Comment by alexrainman** [comment]
Infinite carousel is on my roadmap. Wait for it.
